apps/dashboard-service/src/dashboard/total-section-header.controller.ts

```diff
@@ -0,0 +1,70 @@
+import { Get, Query, Controller } from "@nestjs/common";
+import { ApiOperation } from "@nestjs/swagger";
+import { DashboardQueryDto } from "./dto/dashboard-query.dto";
+import { JsonApiResponse } from "@terramatch-microservices/common/decorators";
+import { CacheService } from "./dto/cache.service";
+import { buildFixedOrderedQueryString, buildJsonApi } from "@terramatch-microservices/common/util/json-api-builder";
+import { TotalSectionHeaderDto } from "./dto/total-serction-header.dto";
+import { DelayedJobDto } from "./delayed-job.dto";
+import { DelayedJob } from "@terramatch-microservices/database/entities";
+
+@Controller("dashboard/v3/totalSectionHeaders")
+export class TotalSectionHeaderController {
+  constructor(private readonly cacheService: CacheService) {}
+
+  @Get()
+  @JsonApiResponse([TotalSectionHeaderDto, DelayedJobDto])
+  @ApiOperation({ summary: "Get total section header" })
+  async getTotalSectionHeader(@Query() query: DashboardQueryDto) {
+    const cacheKey = `dashboard:total-section-header|${this.cacheService.getCacheKeyFromQuery(query)}`;
+    const lastUpdatedAt = await this.cacheService.getTimestampForTotalSectionHeader(
+      this.cacheService.getCacheKeyFromQuery(query)
+    );
+    const cachedData = await this.cacheService.get(cacheKey);
+    const parseCachedData = typeof cachedData === "string" ? JSON.parse(cachedData) : cachedData;
```

Comment:
I think this kind of conversion would better handled in the cache service. The controller shouldn't need to know that sometimes a string comes out of the service.